05-JWT-Basics/starter/middleware/auth.js

```diff
@@ -0,0 +1,20 @@
+const jwt = require("jsonwebtoken")
+const UnauthenticatedError = require("../errors/unauthenticated")
+
+const authenticationMiddleware = async (req, res, next) => {
+  const authHeader = req.headers.authorization
+  if (!authHeader || !authHeader.startsWith("Bearer")) {
```

Comment:
extra challenge: can we rewrite this bit to extract the token with a regular expression? It will be a simple one-liner!

Reply:
I wrote a very simple regex to match the token format, will push new code to update PR. Please let me know if it could be written better? I was playing with the jtw.io and adding or changing new fields would increase the number of characters so I could not set an exact number.

Reply:
So, think about the pattern we're trying to match. 

```
Bearer hrjekfnekfkekrkfjjr
```

We have three things here

1. The "Bearer" prefix, which always stays the same. 
2. A single whitespace " "
3. A string of one or more characters (the token itself)

So, we can describe that pattern with this regex:

```
/Bearer (\w)+/
```

Once you have the regex, you can use it to validate the input and extract the token in one step. Give it a try, let me know how it goes!